### PR TITLE
Prevent fly-by fixture from powering containment field generator

### LINKED ISSUE #29222: Containment field generators are powered by fly-by fixture
## Description
Containment field generators are powered when an entity with the EmitterBolt tag collides with them. However, emitter bolts also have a fly-by fixture that extends far beyond the bolt itself, which collides with the generator, powering it indirectly. I believe this to be unintended behaviour.

**Reproduction**
1. Set up a containment field generator.
2. Point an emitter at a tile *next to* the field generator.
3. Enable the emitter, and watch the containment field generator charge up despite the bolt whizzing past it.

**Screenshots**
image

**Additional context**
I observed this in a recent round on Salamander as part of a tesla setup that only used a single emitter. This behaviour is definitely being used in the wild.

### PR DESCRIPTION
**Pull request by arimah**
## About the PR
Fixes #29222 

TL;DR: the fly-by fixture on the emitter bolt can power the containment field generator.

## Why / Balance
I believe the original behaviour to be unintended, and I personally witnessed it being relied on in a round on Salamander to power a Tesla containment field with only a single emitter.

## Technical details
Adds a field to `ContainmentFieldGeneratorComponent`, `SourceFixtureId`, so we can generate power only when the emitter bolt's `"projectile"` fixture collides with the generator. Otherwise we also catch the `"fly-by"` fixture.

## Media
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
It might break your tesla meta, but should not have any knock-on effects.

**Changelog**
:cl:
- fix: Containment field generators can no longer be powered by an emitter bolt flying past them. The bolt must hit them directly.

### CODE REVIEW

File: Content.Shared/Singularity/Components/ContainmentFieldGeneratorComponent.cs [hunk 1]
```diff
@@ -69,6 +69,14 @@ public int PowerBuffer
     [DataField("idTag", customTypeSerializer: typeof(PrototypeIdSerializer<TagPrototype>))]
     public string IDTag = "EmitterBolt";
 
+    /// <summary>
+    /// Which fixture ID should test collision with from the entity that powers the generator?
+    /// Prevents the generator from being powered by fly-by fixtures.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField]
```
Comment (EmoGarbage404) [inline Content.Shared/Singularity/Components/ContainmentFieldGeneratorComponent.cs:77]: Remove the VV as its applied automatically.

File: Content.Server/Singularity/EntitySystems/ContainmentFieldGeneratorSystem.cs [hunk 2]
```diff
@@ -65,7 +65,8 @@ public override void Update(float frameTime)
     /// </summary>
     private void HandleGeneratorCollide(Entity<ContainmentFieldGeneratorComponent> generator, ref StartCollideEvent args)
     {
-        if (_tags.HasTag(args.OtherEntity, generator.Comp.IDTag))
+        if (_tags.HasTag(args.OtherEntity, generator.Comp.IDTag) &&
```
Comment (EmoGarbage404) [inline Content.Server/Singularity/EntitySystems/ContainmentFieldGeneratorSystem.cs:68]: Flip the order of these checks for mini perf boost

**Review by EmoGarbage404 — APPROVED** [review]

File: Content.Shared/Singularity/Components/ContainmentFieldGeneratorComponent.cs (hunk 1 above)
Comment (arimah) [reply Content.Shared/Singularity/Components/ContainmentFieldGeneratorComponent.cs:76]: Fixed!

File: Content.Server/Singularity/EntitySystems/ContainmentFieldGeneratorSystem.cs (hunk 2 above)
Comment (arimah) [reply Content.Server/Singularity/EntitySystems/ContainmentFieldGeneratorSystem.cs]: I doubt it'll shave off more than a few dozen nanoseconds in practice, but sure, let's go for the nano perf boost. Fixed.

**Comment by MACMAN2003** [comment]
what makes you think this behavior was unintended?

**Comment by EmoGarbage404** [comment]
> what makes you think this behavior was unintended?

well, when something hits something that is a whole tile away and not in the trajectory, one could consider that to be pretty weird and unintended

**Comment by MACMAN2003** [comment]
considering that emitter lasers have a secondary collision thingy that powered the pylons i don't think it was unintended, whoever made emitter bolts wouldn't add a secondary collision thingy for no reason.

**Comment by EmoGarbage404** [comment]
> considering that emitter lasers have a secondary collision thingy that powered the pylons i don't think it was unintended

the reason it powers them is because it does a very dumb blanket collision subscription without checking a fixture ID, an extremely common projectile bug that has happened countless times.

> whoever made emitter bolts wouldn't add a secondary collision thingy for no reason

you're right! which is why, just like every single other fly-by fixture in the codebase, it exists to play the sfx of a bullet whizzing by. This is well known because basically every single projectile has this fixture and they all make checks to exclude it because it's not meant to actually count as a projectile collision.

even ignoring the intent, it makes 0 sense for a laser to collide with things when there is clearly no visible collision occurring with the sprite, so this entire tangent doesn't even matter to begin with.

**Comment by Simyon264** [comment]
Fixes #25528
